src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        if (type == Bulldozer.class) {
+            machines = new BulldozerProducer().get();
+            return new ArrayList<>(machines);
+        }
+        if (type == Excavator.class) {
+            machines = new ExcavatorProducer().get();
+            return new ArrayList<>(machines);
+        }
+        if (type == Truck.class) {
+            machines = new TruckProducer().get();
+            return new ArrayList<>(machines);
+        }
+        return null;
```

Comment:
```suggestion
        return Collections.EMPTY_LIST;
```
it's better to return empty list instead of null to avoid NPE in future